pkg/telemetryquerylanguage/tql/parser.go

```diff
@@ -150,25 +190,38 @@ func parseQuery(raw string) (*ParsedQuery, error) {
 	return parsed, nil
 }
 
-// newParser returns a parser that can be used to read a string into a ParsedQuery. An error will be returned if the string
-// is not formatted for the DSL.
-func newParser() *participle.Parser {
-	lex := lexer.MustSimple([]lexer.SimpleRule{
-		{Name: `Ident`, Pattern: `[a-zA-Z_][a-zA-Z0-9_]*`},
+// buildLexer constructs a SimpleLexer definition.
+// Note that the ordering of these rules matters.
+// It's in a separate function so it can be easily tested alone (see lexer_test.go).
+func buildLexer() *lexer.StatefulDefinition {
+	return lexer.MustSimple([]lexer.SimpleRule{
 		{Name: `Bytes`, Pattern: `0x[a-fA-F0-9]+`},
 		{Name: `Float`, Pattern: `[-+]?\d*\.\d+([eE][-+]?\d+)?`},
 		{Name: `Int`, Pattern: `[-+]?\d+`},
 		{Name: `String`, Pattern: `"(\\"|[^"])*"`},
-		{Name: `Operators`, Pattern: `==|!=|[,.()\[\]]`},
+		{Name: `OpOr`, Pattern: `\b(or)\b`},
+		{Name: `OpAnd`, Pattern: `\b(and)\b`},
+		{Name: `OpEq`, Pattern: `==|!=`},
```

Comment:
Instead of `OpEq` could we do something like `OpComparison`. In the future when we add `<`, `>`, `<=`, and `>=` we'll add them to this token.